# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1266773826 0
#      Sun Feb 21 17:37:06 2010 +0000
# Node ID 2ef5144ccb0de5d569ef19e9e4149b013dd3745f
# Parent  7f6b4c6838eafb4c86272623df151ddf7dc18bc6
Add few OCTINTERP_API tags.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2010-02-21  Michael Goffioul  <michael.goffioul@gmail.com>
 
+	* ov-bool.h (class octave_bool): Add OCTINTERP_API.
+	* unwind-prot.h (class unwind_protect_safe): Ditto.
+
 	* Makefile.am: Add -bindir flag to liboctinterp_la_LDFLAGS. Remove
 	temporary libtool module file after copying oct file.
 
 2010-02-19  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/qr.cc: Increase tolerance for test.
 
 2010-02-19  Jaroslav Hajek  <highegg@gmail.com>
diff --git a/src/ov-bool.h b/src/ov-bool.h
--- a/src/ov-bool.h
+++ b/src/ov-bool.h
@@ -43,16 +43,17 @@ along with Octave; see the file COPYING.
 class Octave_map;
 class octave_value_list;
 
 class tree_walker;
 
 // Real scalar values.
 
 class
+OCTINTERP_API
 octave_bool : public octave_base_scalar<bool>
 {
 public:
 
   octave_bool (void)
     : octave_base_scalar<bool> (false) { }
 
   octave_bool (bool b)
diff --git a/src/unwind-prot.h b/src/unwind-prot.h
--- a/src/unwind-prot.h
+++ b/src/unwind-prot.h
@@ -245,17 +245,19 @@ private:
 
   elem *head;
 };
 
 // Like unwind_protect, but this one will guard against the possibility of seeing
 // an exception (or interrupt) in the cleanup actions. Not that we can do much about
 // it, but at least we won't crash.
 
-class unwind_protect_safe : public unwind_protect
+class
+OCTINTERP_API
+unwind_protect_safe : public unwind_protect
 {
   static void gripe_exception (void);
 
 public:
   ~unwind_protect_safe (void)
     {
       while (*this)
         {
